lib/src/observable_map.dart

```diff
@@ -155,65 +155,37 @@ class ObservableMap<K, V> extends Observable implements Map<K, V> {
   String toString() => MapBase.mapToString(this);
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
-  Map<K2, V2> cast<K2, V2>() {
-    throw new UnimplementedError("cast");
-  }
+  Map<K2, V2> cast<K2, V2>() => _map.cast<K2, V2>();
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
-  Map<K2, V2> retype<K2, V2>() {
-    throw new UnimplementedError("retype");
-  }
+  Map<K2, V2> retype<K2, V2>() => _map.retype<K2, V2>();
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_getter
-  Iterable<Null> get entries {
-    // Change Iterable<Null> to Iterable<MapEntry<K, V>> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("entries");
-  }
+  Iterable<MapEntry<K, V>> get entries => _map.entries;
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
-  void addEntries(Iterable<Object> entries) {
-    // Change Iterable<Object> to Iterable<MapEntry<K, V>> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("addEntries");
+  void addEntries(Iterable<MapEntry<K, V>> entries) {
+    _map.addEntries(entries);
   }
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
-  Map<K2, V2> map<K2, V2>(Object transform(K key, V value)) {
-    // Change Object to MapEntry<K2, V2> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("map");
+  Map<K2, V2> map<K2, V2>(MapEntry<K2, V2> transform(K key, V value)) {
```

Comment:
I think this is OK, though. `List.map()` returns `Iterable<T>`, not `List<T>`. Also, I think `LinkedHashMap.map()` and `HashMap.map()` return an unspecified Map type.

Reply:
Thx